consensus/src/experimental/commit_phase.rs

```diff
@@ -385,17 +433,23 @@ impl CommitPhase {
                 );
             }
 
-            if let Some((blocks, ordered_ledger_info, callback)) =
-                self.commit_channel_recv.next().await
-            {
-                report_err!(
-                    // receive new blocks from execution phase
-                    self.process_executed_blocks(blocks, ordered_ledger_info, callback)
-                        .await,
-                    "Error in processing received blocks"
-                );
-            } else {
-                break;
+            select! {
+                CommitChannelType(blocks, ordered_ledger_info, callback) = self.commit_channel_recv.select_next_some() => {
```

Comment:
maybe we can unify the two select with tokio::select + preconditions

tokio::select! {
  .. = self.commit_channel_recv.next(), if self.block.is_none() => {
  }
  .. = self.commit_msg_rx.next(), if self.block.is_some() => {
  }
}

Reply:
let me try

Reply:
resolved